internal/controller/sync_controller.go

```diff
@@ -0,0 +1,237 @@
+	// if err != nil {
+	// 	return ctrl.Result{}, err
+	// }
+	// objectStorageBackupList, err := objectStorage.ListBackups()
+	// if err != nil {
+	// 	return ctrl.Result{}, nil
+	// }
+
+	// // get all Velero backups that
+	// // NAC owns
+	// // are finished
+	// // were created from same namespace as NABSL
+	// var veleroBackupList velerov1.BackupList
+	// labelSelector := client.MatchingLabels(function.GetNonAdminLabels())
+
+	// if err := r.List(ctx, &veleroBackupList, client.InNamespace(r.OADPNamespace), labelSelector); err != nil {
+	// 	return ctrl.Result{}, err
+	// }
+	// logger.V(1).Info(fmt.Sprintf("%v Backups owned by NAC controller in OADP namespace", len(veleroBackupList.Items)))
+
+	// var backupsToSync []velerov1.Backup
+	// // var possibleBackupsToSync []velerov1.Backup
+	// var possibleBackupsToSync []string
+	// for _, backup := range veleroBackupList.Items {
+	// 	if backup.Status.CompletionTimestamp != nil &&
+	// 		backup.Annotations[constant.NabOriginNamespaceAnnotation] == nonAdminBackupStorageLocation.Namespace {
+	// 		possibleBackupsToSync = append(possibleBackupsToSync, backup.Name)
+	// 	}
+	// }
+
+	// objectStorageBackupSet := sets.New(objectStorageBackupList...)
+	// inClusterBackupSet := sets.New(possibleBackupsToSync...)
+	// backupsToSyncSet := objectStorageBackupSet.Difference(inClusterBackupSet)
+
+	// for backupName := range backupsToSyncSet {
+	// 	veleroBackup := velerov1.Backup{}
+	// 	err := r.Get(ctx, types.NamespacedName{
+	// 		Namespace: r.OADPNamespace,
+	// 		Name:      backupName,
+	// 	}, &veleroBackup)
+	// 	if err != nil {
+	// 		if apierrors.IsNotFound(err) {
+	// 			// not yet synced by velero
+	// 			continue
+	// 		}
+	// 		logger.Error(err, "Unable to fetch Velero Backup")
+	// 		return ctrl.Result{}, err
+	// 	}
+	// 	backupsToSync = append(backupsToSync, veleroBackup)
+	// }
+
+	// logger.V(1).Info(fmt.Sprintf("%v Backup(s) to sync to NonAdminBackupStorageLocation namespace", len(backupsToSync)))
+	// for _, backup := range backupsToSync {
+	// 	nab := &nacv1alpha1.NonAdminBackup{
+	// 		ObjectMeta: v1.ObjectMeta{
+	// 			Name:      backup.Annotations[constant.NabOriginNameAnnotation],
+	// 			Namespace: backup.Annotations[constant.NabOriginNamespaceAnnotation],
+	// 			// TODO sync operation does not preserve labels
+	// 			Labels: map[string]string{
+	// 				"openshift.io/oadp-nab-synced-from-nacuuid": backup.Labels[constant.NabOriginNACUUIDLabel],
+	// 			},
+	// 		},
+	// 		Spec: nacv1alpha1.NonAdminBackupSpec{
+	// 			BackupSpec: &backup.Spec,
+	// 		},
+	// 	}
+	// 	nab.Spec.BackupSpec.StorageLocation = nonAdminBackupStorageLocation.Name
+	// 	r.Create(ctx, nab)
+	// }
+
+	// ------------------------------------------------------------------------
+
+	// OPTION 2: compare BSLs specs
+	// PROBLEM: if user deletes NABSL, and them recreates it, velero backup will point to a BSL that does not exist
```

Comment:
I would try this approach, but it is not perfect

We could document that there are limitations, if there no other options :(